python_modules/dagster/dagster/_core/definitions/unresolved_asset_job_definition.py

```diff
@@ -182,9 +182,8 @@ def resolve(
     ) -> "JobDefinition":
         """Resolve this UnresolvedAssetJobDefinition into a JobDefinition."""
         assets = asset_graph.assets_defs
-        source_assets = asset_graph.source_assets
         selected_asset_keys = self.selection.resolve(asset_graph)
-        if asset_graph.includes_materializable_and_source_assets(selected_asset_keys):
+        if asset_graph.includes_materializable_and_external_assets(selected_asset_keys):
```

Comment:
not your problem, but this is pretty goofy. Aren't we supporting observable externals and materializables in the same job at this point?